android/Staccato_AN/app/src/main/java/com/on/staccato/presentation/bindingadapter/TextBindingAdapter.kt

```diff
@@ -0,0 +1,158 @@
+import androidx.databinding.BindingAdapter
+import com.on.staccato.R
+import com.on.staccato.domain.model.MemoryCandidate
+import com.on.staccato.domain.model.MemoryCandidates
+import com.on.staccato.presentation.common.getFormattedLocalDateTime
+import okhttp3.internal.format
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+@BindingAdapter(value = ["selectedMemory", "memoryCandidates"])
+fun TextView.setSelectedMemory(
+    selectedMemory: MemoryCandidate?,
+    memoryCandidates: MemoryCandidates?,
+) {
+    if (memoryCandidates?.memoryCandidate?.isEmpty() == true) {
+        text = resources.getString(R.string.visit_creation_no_memory_hint)
+        setTextColor(resources.getColor(R.color.gray3, null))
+        isClickable = false
+        isFocusable = false
+    } else if (selectedMemory == null) {
+        text = resources.getString(R.string.visit_creation_memory_selection_hint)
+        setTextColor(resources.getColor(R.color.gray3, null))
+        isClickable = true
+        isFocusable = true
+    } else {
+        text = selectedMemory.memoryTitle
+        setTextColor(resources.getColor(R.color.staccato_black, null))
+    }
+}
+
+@BindingAdapter(value = ["dateTimeWithAmPm", "memoryCandidates"])
+fun TextView.setDateTimeWithAmPm(
+    dateTime: LocalDateTime?,
+    memoryCandidates: MemoryCandidates?,
+) {
+    if (memoryCandidates?.memoryCandidate?.isEmpty() == true) {
+        text = resources.getString(R.string.visit_creation_memory_selection_hint)
+        setTextColor(resources.getColor(R.color.gray3, null))
+        isClickable = false
+        isFocusable = false
+    } else {
+        text = dateTime?.let(::getFormattedLocalDateTime) ?: ""
+        setTextColor(resources.getColor(R.color.staccato_black, null))
+        isClickable = true
+        isFocusable = true
+    }
+}
+
+@BindingAdapter(
+    value = ["startDate", "endDate"],
+)
+fun TextView.setMemoryPeriod(
+    startDate: LocalDate?,
+    endDate: LocalDate?,
+) {
+    if (startDate == null || endDate == null) {
+        text = resources.getString(R.string.memory_creation_period_hint)
+        setTextColor(resources.getColor(R.color.gray3, null))
+    } else {
+        text =
+            resources.getString(R.string.memory_creation_selected_period)
+                .format(startDate, endDate)
+        setTextColor(resources.getColor(R.color.staccato_black, null))
+    }
+}
+
+@BindingAdapter("isMemoryCandidatesEmpty")
+fun TextView.setIsMemoryCandidatesEmptyVisibility(memoryCandidates: MemoryCandidates?) {
+    isGone = !memoryCandidates?.memoryCandidate.isNullOrEmpty()
+}
+
+@BindingAdapter("isMemoryEmpty")
+fun TextView.setIsMemoryEmptyVisibility(items: List<Int>?) {
+    isGone = !items.isNullOrEmpty()
+}
+
+@BindingAdapter("visitedAtHistory")
+fun TextView.formatVisitedAtHistory(visitedAt: LocalDateTime?) {
+    text = visitedAt?.let {
+        getFormattedLocalDateTime(it) + resources.getString(R.string.visit_at_history)
```

Comment:
```suggestion
@BindingAdapter("visitedAtHistory")
fun TextView.formatVisitedAtHistory(visitedAt: LocalDateTime?) {
    text = visitedAt?.let {
         resources.getString(R.string.visit_at_history).format(getFormattedLocalDateTime(it))
```

```xml
    // fragment_visit
    <string name="visit_at_history">%s에 방문했어요</string>
```
`visit_at_history`에 `%s`를 추가하고 format을 활용하는 건 어떠신가용?

Reply:
반영했습니다!